src/ConstructiveSolidGeometry/LinePrimitives/Ellipse.jl

```diff
@@ -0,0 +1,68 @@
+"""
+    struct Ellipse{T,TR} <: AbstractLinePrimitive{T}
+
+* `r::TR`: 
+    * TR = Real -> Circle (a = b = r)
+    * TR = (Real, Real) -> Circular Annulus (r_in = r[1], r_out = r[2])
+    * TR = ((Real,), (Real,)) -> Ellipse (a = r[1][1], b = r[2][1])
+    * TR = ((Real, Real),(Real, Real)) -> Elliptical Annulus \n(a_in = r[1][1], a_out = r[1][2], b_in = r[2][1], b_out = r[2][2])
+    * Not all are implemented yet
+
+* `φ::TP`: 
+    * TP = Nothing <-> Full in φ
+    * ...
+"""
+@with_kw struct Ellipse{T,TR,TP} <: AbstractLinePrimitive{T}
+    r::TR = 1
+    φ::TP = nothing
+
+    origin::CartesianPoint{T} = zero(CartesianPoint{T})
+    rotation::SMatrix{3,3,T,9} = one(SMatrix{3, 3, T, 9})
+end
+
+Ellipse{T,TR,TP}( e::Ellipse{T,TR,TP}; 
+            origin::CartesianPoint{T} = b.origin,
+            rotation::SMatrix{3,3,T,9} = b.rotation) where {T,TR,TP} =
```

Comment:
should be `e.origin` and `e.rotation` by default

Reply:
Ah, yes, but actually I don't need that constructor as it is only needed for Volume Primitives (Close <-> Open)